api/autopilot.go

```diff
@@ -77,6 +93,96 @@ type (
 	}
 )
 
+func (hgb HostGougingBreakdown) CanDownload() (errs []error) {
+	for _, err := range []error{
```

Comment:
For `CanDownload`, `CanUpload` and `CanForm` I wasn't too sure whether we should include more checks than strictly necessary or try to keep it to the minimum. I went with always including both versions of the gouging and contract checks, but for uploads and downloads just select v2 and v3 respectively.

Reply:
I think we might be fine just doing a full gouging check every time and not distinguish between download, upload etc. Because we should have enough redundancy to not have to rely on downloading from one or a few gouging hosts.

Reply:
I will F/U with that, I'm thinking if we simplify this we should probably add a couple of lines to the download code that uses hosts that are gouging if it means the file could otherwise not be downloaded